### ISSUE TITLE
ChaiScript implicit function conversion failure

### ISSUE THREAD

**Issue by stephenberry** [ChaiScript/ChaiScript#361]
* Compiler: MSVC 2015
 * Operating System: Windows 10
 * Architecture: x64 
 
This issue caused me significant headache to find. But, here is a simple example:

``` C++
using system_t = std::function<void(double, double, double)>;

void evaluator(const system_t& system, double x, double y, double z) {}

int main()
{
   chaiscript::ChaiScript chai;

   chai.add(chaiscript::user_type<system_t>(), "system_t");

   chai.add(chaiscript::fun(&evaluator), "evaluator");
   . . .
```
Script:
```
auto x = 1.0
auto y = 2.0
auto z = 3.0

auto system = fun(x, y, z) {}

evaluator(system, x, y, z)
```
This code doesn't do anything, but it doesn't throw any exceptions as expected.

Breaking changes: add a System object and register a type conversion to a std::function
``` C++
struct System
{
   void operator()(double x, double y, double z) {}
};

// in main:
chai.add(chaiscript::type_conversion<System, system_t>());
```

Attempting to run the same script now produces the error:  
"Error with function dispatch with function `evaluator` . . .

Essentially the type conversion for the ChaiScript lambda to std::function can no longer be made. I'm guessing because the type conversion from System is overriding it.

This is a significant issue for me right now, so any advice in fixing it would be great!

**Comment by lefticus** [comment]
I'm sorry to just now be getting to this issue. Is it still presenting a problem for you?

**Comment by stephenberry** [comment]
@lefticus Yes, we've had to remove some implicit conversions so that ChaiScript functions can be passed to C++ code. Do you have an idea of what is causing this issue?
